src/Mailgun/Api/MailingList.php

```diff
@@ -0,0 +1,268 @@
+    }
+
+    /**
+     * Updates a mailing list.
+     *
+     * @param string $address    Address of the mailing list
+     * @param array  $parameters Array of field => value pairs to update
+     *
+     * @return UpdateResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function update($address, $parameters = [])
+    {
+        Assert::stringNotEmpty($address);
+        Assert::isArray($parameters);
+
+        foreach ($parameters as $field => $value) {
+            switch ($field) {
+                case 'access_level':
+                    Assert::oneOf($value, ['readonly', 'members', 'everyone']);
+
+                    break;
+            }
+        }
+
+        $response = $this->httpPut(sprintf('/v3/lists/%s', $address), $parameters);
+
+        return $this->hydrateResponse($response, UpdateResponse::class);
+    }
+
+    /**
+     * Removes a mailing list from the domain.
+     *
+     * @param string $address Address of the mailing list
+     *
+     * @return DeleteResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function delete($address)
+    {
+        Assert::stringNotEmpty($address);
+
+        $response = $this->httpDelete(sprintf('/v3/lists/%s', $address));
+
+        return $this->hydrateResponse($response, DeleteResponse::class);
+    }
+
+    /**
+     * Returns a paginated list of members of the mailing list.
+     *
+     * @param string $address    Address of the mailing list
+     * @param int    $limit      Maximum number of records to return (optional: 100 by default)
+     * @param mixed  $subscribed `yes` to lists subscribed, `no` for unsubscribed. list all if null
+     *
+     * @return MembersResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function members($address, $limit = 100, $subscribed = null)
+    {
+        Assert::stringNotEmpty($address);
+        Assert::integer($limit);
+        Assert::oneOf($subscribed, [null, 'yes', 'no']);
+
+        $params = compact('limit', 'subscribed');
+
+        $response = $this->httpGet(sprintf('/v3/lists/%s/members/pages', $address), $params);
+
+        return $this->hydrateResponse($response, MembersResponse::class);
+    }
+
+    /**
+     * Shows a single member of the mailing list.
+     *
+     * @param string $list    Address of the mailing list
+     * @param string $address Address of the member
+     *
+     * @return ShowMemberResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function showMember($list, $address)
+    {
+        Assert::stringNotEmpty($list);
+        Assert::stringNotEmpty($address);
+
+        $response = $this->httpGet(sprintf('/v3/lists/%s/members/%s', $list, $address));
+
+        return $this->hydrateResponse($response, ShowMemberResponse::class);
+    }
+
+    /**
+     * Creates (or updates) a member of the mailing list.
```

Comment:
`createMember` through `POST` HTTP Method is just to create - not update - records. Updates are managed through the `PUT` HTTP Method and the `updateMember` method.

Reply:
> but - in the end - the record is not added because it already exists.

The record actually get's updated in the process and the Response is a valid CreateMemberResponse with all the details about the "added" Member. I am well aware of the REST principles, but iam not sure how to handle this any other way - unless we drop the `$upsert` Parameter and don't allow this `feature` of the Mailgun API. If we do allow it, then we also have to mention it in the doc blocks, right? :) In the end, Mailgun violates the principles, not us :P

Reply:
🤔 

@Nyholm what do you think? In fact, both points are good...